File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sumDict = {}
        for i in range(len(nums)):
            cur = target - nums[i]

            if cur in sumDict:
                return [sumDict[cur], i]
            else:
                sumDict[nums[i]] = i

File: Data Structures & Algorithms/duplicate-integer/submission-3.swift
class Solution {
    func hasDuplicate(_ nums: [Int]) -> Bool {
        var dupDict: [Int : Int] = [:]
        for num in nums {
            if (dupDict[num] != nil){
                return true
            } else {
                dupDict[num] = 1
            }
        }
        return false
    }
}
